src/hooks/useTodos.ts

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useRef, useState } from 'react';
+import { Todo } from '../types/types';
+import { getTodos } from '../api/todos';
+import { ErrorMessages } from '../types/enums';
+import { useErrorReset } from './useErrorReset';
+import { useInputFocus } from './useInputFocus';
+import { useDeleteTodos } from './useDeleteTodos';
+import { useEditTodos } from './useEditTodos';
+import { useAddTodos } from './useAddTodos';
+import { useUpdateTodos } from './useUpdateTodos';
+
+export const useTodos = () => {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoadingTodos, setIsLoadingTodos] = useState<boolean>(false);
+  const { currentError, setCurrentError } = useErrorReset();
+  const { isLoadingDelete, handleDeleteTodos, handleDeleteAllTodos } =
+    useDeleteTodos(setTodos, setCurrentError);
+  const {
+    isLoadingEdit,
+    handleEditTodo,
+    editTodoQuery,
+    handleEditTodoQuery,
+    defaultEditInputValue,
+    handleActivateEdit,
+    activeEdit,
+    setActiveEdit,
+    setEditTodoQuery,
+  } = useEditTodos(handleDeleteTodos, setTodos, setCurrentError, todos);
+  const { isLoadingAdd, tempTodo, handleAddTodo, setTempTodo } = useAddTodos(
+    setCurrentError,
+    setTodos,
+    inputRef,
+  );
+  const { isLoadingUpdate, handleCheckTodo, handleToggleAll } = useUpdateTodos(
+    todos,
+    setTodos,
+    setCurrentError,
+    setIsLoadingTodos,
+  );
```

Comment:
to be honest it's hard to understand what happened here
wat is the diff gor exemple here
handleEditTodo,
    editTodoQuery,
    handleEditTodoQuery,
    defaultEditInputValue,
    handleActivateEdit,
    activeEdit,
    setActiveEdit,
    
    
    so many states is there a chance to get rid of smth?